### DISCUSSION TITLE
Does qdrant support BM25-like keyword search?

### DISCUSSION THREAD

**Discussion by Bhavnick-Yali** [qdrant/qdrant#971] [General]
Hey! :ice_cube: 
I am new to qdrant and don't know it's functionality that well. 

I am trying to build a hybrid vector search + keyword search engine and was wondering if qdrant supports BM25 search as well alongside it's vector search. There are some queries that require a keyword search because of which such support is important for my application. 

Thanks.

**Comment by generall** [comment]
Hi @Bhavnick-Yali! Qdrant will support full-text filtering in the next release, however it won't be a full-fledged BM25 search. The main problem here is combine BM25 score with similarity scores, which is usually non-trivial task, which usually requires custom re-ranking model. In order to achieve best performance and accuracy, I would recommend to use specialized service for full-text search. Like https://github.com/meilisearch/meilisearch for example. In this case you would be able to flexibly decide how to mix results and won't be limited in the choice of model for rescoring
